lib/pure/asyncdispatch.nim

```diff
@@ -160,6 +221,41 @@
 ## graph.
 ##
 ##
+## Multisync
+## =========
+##
+## The ``{.multisync.}`` Pragma is similar to the ``{.async.}`` Pragma,
+## ``{.multisync.}`` Pragma converts a procedure to an asynchronous *and* synchronous procedure,
+## then the same function can be used on asynchronous *and* synchronous code,
+## you can think of it as a function that is async or sync depending on how you use it.
+##
+## The usage and features are the same of ``{.async.}`` (mentioned above). Example:
+##
+## .. code-block:: nim
+##   import asyncdispatch, httpclient
+##
+##   type
+##     Client = HttpClient
+##     AsyncClient = AsyncHttpClient
+##
+##   proc multisync_function(this: Client | AsyncClient): Future[string] {.multisync.} =
+##     const url = "https://nim-lang.org"
+##     result =
+##       when this is AsyncClient: await newAsyncHttpClient().getContent(url)
+##       else: newHttpClient().getContent(url)
```

Comment:
This is a bad example, create the http client and assign it to a variable instead